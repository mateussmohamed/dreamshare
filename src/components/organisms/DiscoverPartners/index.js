import React from 'react'
import styled from 'styled-components'

import { Container, Row, Column } from 'atoms'
import { Button, Partner } from 'molecules'

const Title = styled.h1`
  ${props => props.theme.typography.montserratStyle};
  font-size: 22px;
  font-weight: bold;
  text-align: center;
  color: ${props => props.theme.colors.greyishBrown};
`

const FlexColumn = styled(Column)`
  @media only screen and (min-width: 320px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`

const Partners = styled.section`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`

const DiscoverPartners = () => (
  <Container m={['20px auto', '40px auto', '100px auto']}>
    <Row justifyContent="center" mb={50}>
      <Title>Meet a partner for your best holiday</Title>
    </Row>
    <Row flexDirection="column" mx={0}>
      <FlexColumn width={[1, 1, 1]}>
        <Partners>
          <Column width={[1, 1 / 2, 1 / 4]} flex="none">
            <Partner
              name="Bradley Hunter"
              bio="Based in Chicago. I love playing tennis and loud music."
              avatar="/assets/images/partner-1.jpg"
              icon="/assets/icons/music.svg"
            />
          </Column>
          <Column width={[1, 1 / 2, 1 / 4]} flex="none">
            <Partner
              name="Bradley Hunter"
              avatar="/assets/images/partner-2.jpg"
              bio="Based in Chicago. I love playing tennis and loud music."
              icon="/assets/icons/brush.svg"
            />
          </Column>
          <Column width={[1, 1 / 2, 1 / 4]} flex="none">
            <Partner
              name="Bradley Hunter"
              avatar="/assets/images/partner-3.jpg"
              bio="Based in Chicago. I love playing tennis and loud music."
              icon="/assets/icons/camera.svg"
            />
          </Column>
          <Column width={[1, 1 / 2, 1 / 4]} flex="none">
            <Partner
              name="Bradley Hunter"
              avatar="/assets/images/partner-4.jpg"
              bio="Based in Chicago. I love playing tennis and loud music."
              icon="/assets/icons/airplane.svg"
            />
          </Column>
        </Partners>
      </FlexColumn>
    </Row>
    <Row justifyContent="center" mt={50}>
      <Button primary>See other partners</Button>
    </Row>
  </Container>
)

export default DiscoverPartners
