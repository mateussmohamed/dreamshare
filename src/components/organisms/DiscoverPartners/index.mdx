---
name: DiscoverPartners
menu: Organisms
---

import { Playground, PropsTable } from 'docz'
import DiscoverPartners from './index'

# DiscoverPartners
<PropsTable of={DiscoverPartners} />

<Playground>
  <DiscoverPartners />
</Playground>
