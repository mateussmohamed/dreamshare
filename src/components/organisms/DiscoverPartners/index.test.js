import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import DiscoverPartners from './index'

describe('[<DiscoverPartners />]', () => {
  const wrap = (props = {}) => shallow(<DiscoverPartners {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({})

    expect(wrapper).toMatchSnapshot()
  })
})
