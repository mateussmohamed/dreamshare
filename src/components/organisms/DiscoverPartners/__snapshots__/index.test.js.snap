// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Hero />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Hero__HeroContainer-sc-1co6wbd-0",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "display: flex;",
          [Function],
          [Function],
          [Function],
          [Function],
          "width:100%;height:100vh;flex-direction:column;",
        ],
      },
      "displayName": "Hero__HeroContainer",
      "propTypes": Object {
        "alignItems": [Function],
        "flexDirection": [Function],
        "flexWrap": [Function],
        "justifyContent": [Function],
      },
      "render": [Function],
      "styledComponentId": "Hero__HeroContainer-sc-1co6wbd-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <HeroImage
    src=""
  />
  <Hero__HeroContent>
    <HeroHeader />
    <HeroSpotlight
      description="hero description"
      title="Hero Title"
    />
  </Hero__HeroContent>
</StyledComponent>
`;
